File: LeetCode/Easy/0168-excel-sheet-column-title/0168-excel-sheet-column-title.cs
public class Solution {
    private const int AlphabetNumericalSystemBase = 26;
    private static readonly int MaxCharsNumForInt = (int)Math.Log(int.MaxValue, AlphabetNumericalSystemBase) + 1;
    public string ConvertToTitle(int columnNumber) {
        Span<char> buffer = stackalloc char[MaxCharsNumForInt];
        var pos = buffer.Length;
        while (columnNumber > 0)
        {
            columnNumber--; // mapping from A->1-based to A->0-based

            pos--;
            buffer[pos] = (char)('A' + columnNumber % AlphabetNumericalSystemBase);

            columnNumber /= AlphabetNumericalSystemBase;
        }

        return new string(buffer[pos..]);
    }
}

File: LeetCode/Easy/0168-excel-sheet-column-title/0168-excel-sheet-column-title.py
class Solution:
    LETTERS_COUNT = 26
    def convertToTitle(self, columnNumber: int) -> str:
        current_col_number_part = columnNumber
        column_title = []

        while current_col_number_part:
            current_col_number_part -= 1
            val = current_col_number_part % self.LETTERS_COUNT
            column_title.append(chr(ord("A") + val))
            current_col_number_part //= self.LETTERS_COUNT

        return "".join(reversed(column_title))
        